hack/test-go.sh

```diff
@@ -303,6 +309,9 @@ reportCoverageToCoveralls() {
   fi
 }
 
+# Start etcd
+kube::etcd::start
```

Comment:
For me it seems that you are starting etcd that is alive while running all unit tests. This doesn't seem to be ideal for me (in particular - unit tests can "communicate" with each other via this etcd instance).
I think that ideally we should start etcd for each unit tests separately (and only for those that needs it - there are not that many of those tests). And even better would be to start it not in a separate process, but just link the etcd code to our tests and start it e.g. as a separate goroutine.

@lavalamp for thoughts